### PR TITLE
Redirect to start page from referrals flow

### PR DESCRIPTION
**Pull request by steventux**
### Context

If the employer form or user accounts features are not active we shouldn't allow users to access the referrals flow.
https://trello.com/c/OelDJYCH/993-redirect-away-from-referrals-flow-when-user-accounts-feature-flag-is-inactive

### Changes proposed in this pull request

- Redirect to start path if either of the employer form or user accounts features are inactive.
- Refactor some common system spec steps into a support module.

### Guidance to review

The last commit isn't relevant to the redirecting, just DRYing up system spec steps

### Checklist

- [x] Attach to Trello card
- [x] Rebased main
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

File: app/controllers/referrals/base_controller.rb [hunk 1]
```diff
@@ -12,10 +17,15 @@ def current_referral
     end
     helper_method :current_referral
 
-    def redirect_if_feature_flag_disabled
-      return if FeatureFlags::FeatureFlag.active?(:employer_form)
+    def redirect_if_feature_flag_disabled(
+      feature_flag:,
+      message: nil,
```
Comment (vassyz) [inline app/controllers/referrals/base_controller.rb:22]: Why did you add the `message` and `path` arguments? They don't seem to be passed when the method gets called.

Comment (steventux) [reply app/controllers/referrals/base_controller.rb]: @vassyz I have remove this and reworked the PR.

File: app/controllers/referrals/base_controller.rb [hunk 2]
```diff
@@ -30,5 +31,12 @@ def store_user_location!
     def go_to_check_answers?
       params["go_to_check_answers"] == "true"
     end
+
+    def redirect_referrals_requests_if_user_accounts_disabled
+      return if request.path !~ %r{^/referral}
```
Comment (malcolmbaig) [inline app/controllers/referrals/base_controller.rb:36]: Is there anything which inherits from this base controller, which doesn't match on the regex? Just thinking if inheriting from this controller is enough to warrant applying this conditional redirect.

File: spec/support/system/common_steps.rb [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  def when_i_click_save_and_continue
+    click_on "Save and continue"
+  end
+  alias_method :and_i_click_save_and_continue, :when_i_click_save_and_continue
+
+  def when_i_click_back
+    click_on "Back"
+  end
+  alias_method :and_i_click_back, :when_i_click_back
+end
```
Comment (malcolmbaig) [inline spec/support/system/common_steps.rb:51]: 👌🏼

**Review by malcolmbaig — APPROVED** [review]
LG
